test/functional/mempool_expiry.py

```diff
@@ -26,44 +26,48 @@
 class MempoolExpiryTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
-
-    def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
+        self.setup_clean_chain = True
 
     def test_transaction_expiry(self, timeout):
         """Tests that a transaction expires after the expiry timeout and its
         children are removed as well."""
         node = self.nodes[0]
+        self.wallet = MiniWallet(node)
+
+        # Ensure the mempool starts empty.
+        node.generate(1)
```

Comment:
Why is this needed? You are generating a block in the next line, which cleares the mempool

Reply:
TLDR; I made everything explicit in this commit https://github.com/bitcoin/bitcoin/pull/20276/commits/4b99dc5ef62fe1d1a6e23d71f55570dc14aac9fa because I realized this is all confusing. Thanks for pointing this out. 

It was because fees from the transactions in the first test run (that are still in the mempool) `self.test_transaction_expiry(DEFAULT_MEMPOOL_EXPIRY)` get included in `MiniWallet`'s utxo set when this runs for the next sub-test `self.test_transaction_expiry(CUSTOM_MEMPOOL_EXPIRY)` if the first block is generated with `self.wallet.generate`

Then because of the way `MiniWallet` sorts UTXOs based on which is largest it means we end up using an expired transaction for the "trigger" transaction, and get a weird error in the 2nd test run, but not the first - which threw me for a loop